common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -186,25 +186,54 @@ def selected_children(self):
         if hasattr(self, "_selected_set"):
             # Already done:
             return self._selected_set  # pylint: disable=access-member-before-definition
+
+        lib_tools = self.runtime.service(self, 'library_tools')
+        format_block_keys = lambda block_keys: lib_tools.create_block_analytics_summary(self.location.course_key, block_keys)
+
         # Determine which of our children we will show:
         selected = set(tuple(k) for k in self.selected)  # set of (block_type, block_id) tuples
         valid_block_keys = set([(c.block_type, c.block_id) for c in self.children])  # pylint: disable=no-member
         # Remove any selected blocks that are no longer valid:
-        selected -= (selected - valid_block_keys)
+        invalid_block_keys = (selected - valid_block_keys)
+        if invalid_block_keys:
+            selected -= invalid_block_keys
+            # Publish an event for analytics purposes:
+            self.runtime.publish(self, "edx.librarycontentblock.content.removed", {
+                "location": unicode(self.location),
+                "blocks": format_block_keys(invalid_block_keys),
+                "reason": "invalid",  # Deleted from library or library being used has changed
+            })
```

Comment:
Can we also add "result" here too?  It's just as useful for removed calls as for added calls.  In the example scenario you gave, where the count is reduced, we only get the removed event.